tensorflow_datasets/core/dataset_info.py

```diff
@@ -408,19 +409,52 @@ def initialize_from_bucket(self):
       gcs_utils.download_gcs_file(fname, out_fname)
     self.read_from_directory(tmp_dir)
 
-  def __str__(self):
-    splits_pprint = "{\n %s\n    }" % (
-        pprint.pformat(
-            {k: self.splits[k] for k in sorted(list(self.splits.keys()))},
-            indent=8, width=1)[1:-1])
-    features_dict = self.features
-    features_pprint = "%s({\n %s\n    }" % (
-        type(features_dict).__name__,
-        pprint.pformat({
-            k: features_dict[k] for k in sorted(list(features_dict.keys()))
-        }, indent=8, width=1)[1:-1])
-    citation_pprint = '"""\n%s\n    """' % "\n".join(
-        [u" " * 8 + line for line in self.citation.split(u"\n")])
+  def show_examples(self, ds, rows=3, cols=3, plot_scale=3):
+    """Display some random (rows*columns) images from the image-dataset
```

Comment:
Could you add some documentation about which assumptions are made ? For which dataset will this works,...

Also update the kwargs descriptions

Reply:
Actually it will work with all `image` datasets, but labels in some datasets will not be displayed.
What do u mean by `kwargs` ? Are u talking about `ds, row, cols, plot_scale` ?

Reply:
This function is available for all datasets, so it is important to clarify that this will only works for images dataset which only have a single main image.

Yes, the current docstring do not match the `ds, row, cols, plot_scale` (where is plot_scale ?).

Reply:
It will works only for image datasets which only have a single main image. Otherwise it will raise that ValueError.